lib/rule-executors.ts

```diff
@@ -10,9 +10,14 @@ export function doEval(expr: string): Promise<boolean> {
   });
 }
 
-export function click(selector: string, all = false): boolean {
-  const elem = document.querySelectorAll<HTMLElement>(selector);
-  enableLogs && console.log("[click]", selector, all, elem);
+export function click(selectorOrElements: string | HTMLElement[], all = false): boolean {
+  let elem: HTMLElement[] = [];
+  if (typeof selectorOrElements === 'string') {
+    elem = Array.from(document.querySelectorAll<HTMLElement>(selectorOrElements));
+  } else {
+    elem = selectorOrElements;
+  }
+  enableLogs && console.log("[click]", selectorOrElements, all, elem);
```

Comment:
I'm not sure that we need to change the `click` API for this use-case. You can just call `t.click()` directly in `sourcepoint-frame.ts`.

Reply:
Direct `.click()` calls are possible, but they make debugging class-based rules more difficult. They don't produce logs, so it's harder to see where we failed at a glance. And they don't return the success result, which adds boilerplate for non-optional steps.

My idea was to make these utility functions be used both in declarative and class-based rules so both syntaxes provide the same functionality